runtime/src/accounts_db.rs

```diff
@@ -1307,28 +1337,104 @@ impl AccountsDB {
             }
         }
 
+        info!(
+            "root counts (generate_index, middle): {}",
+            accounts_index.roots.len()
+        );
+
         let mut counts = HashMap::new();
         for slot_list in accounts_index.account_maps.values() {
-            for (_slot, account_entry) in slot_list.read().unwrap().iter() {
+            for (slot, account_entry) in slot_list.read().unwrap().iter() {
+                trace!("found while conting: slot: {}  {:?}", slot, account_entry);
                 *counts.entry(account_entry.store_id).or_insert(0) += 1;
             }
         }
-        for slot_stores in storage.0.values() {
+        let mut dead_slots = vec![];
+        let mut alive_slots = vec![];
+        for (slot, slot_stores) in storage.0.iter() {
+            let mut is_dead = true;
+            for id in slot_stores.keys() {
+                if counts.get(&id).is_some() {
+                    is_dead = false;
+                }
+            }
             for (id, store) in slot_stores {
-                if let Some(count) = counts.get(&id) {
+                if !is_dead {
+                    let full_count = store.accounts.accounts(0).len();
                     trace!(
-                        "id: {} setting count: {} cur: {}",
+                        "slot: {} id: {} setting cur: {} full_count: {}",
+                        slot,
                         id,
-                        count,
-                        store.count_and_status.read().unwrap().0
+                        store.count_and_status.read().unwrap().0,
+                        full_count,
                     );
-                    store.count_and_status.write().unwrap().0 = *count;
+                    store.count_and_status.write().unwrap().0 = full_count;
                 } else {
-                    trace!("id: {} clearing count", id);
-                    store.count_and_status.write().unwrap().0 = 0;
+                    let mut c = store.count_and_status.write().unwrap();
```

Comment:
This part is the main dish in this PR; In this condition, we're pretty sure we can remove those Storages (=AppendVecs).